### ISSUE TITLE
Jackon Afterburner Module Registration Prevents Package Property Deserialization

### ISSUE THREAD

**Issue by gilbode** [dropwizard/dropwizard#1145]
Dropwizard registers the jackson Afterburner module but doesn't give any way to customize its configuration.  The Afterburner module provides the property _cfgUseValueClassLoader which can be set so that afterburner can deserialize protected / package visible properties.  We've got some classes with package visible properties and when we try to serialize/deserialize them we get all sorts of warnings like below:

 WARNING: Disabling Afterburner serialization for type class MyClass, field #1, due to access error (type java.lang.IllegalAccessError, message=mypackage/MyClass)
17:41:19.071 [ERROR] [system.err]
17:41:19.071 [ERROR] [system.err] java.lang.IllegalAccessError: mypackage/MyClass

Seems that dropwizard should either allow end users to control whether the after burner module is loaded or not via configuration and/or allow setting of its configuration properties.

**Comment by nickbabcock** [comment]
I did some digging and [according to the documentation](http://fasterxml.github.io/jackson-module-afterburner/javadoc/2.4.0/com/fasterxml/jackson/module/afterburner/AfterburnerModule.html), the property should be on by default. And Dropwizard doesn't change that configuration. I'm curious how this error comes about.

**Comment by arteam** [comment]
In the upcoming 0.9.0 there will be support for setting a custom `ObjectMapper` without the `Afterburner` module during the bootstrap phase of an application. See #1112.

**Comment by gilbode** [comment]
Thanks for looking into this, I'm wondering if maybe the afterburner module was not the culprit.  I was running my app from gradle where the classpath consisted of jar files  in the local cache, directories etc. (i.e. different code sources).  

When we build our app we create a single JAR file and the warnings go away.  I had suspected it was due to the afterburner settings done by dropwizard but wonder if there is some further limitation of having multiple code sources.  It isn't immediately obvious to me what is actually causing the issue however.  All of my application classes are in a single directory code source on the classpath at the time I was encountering the error.
